server/utils.go

```diff
@@ -279,3 +280,24 @@ func getUlimitsFromConfig(config Config) ([]ulimit, error) {
 func translateLabelsToDescription(labels map[string]string) string {
 	return fmt.Sprintf("%s/%s/%s", labels[types.KubernetesPodNamespaceLabel], labels[types.KubernetesPodNameLabel], labels[types.KubernetesContainerNameLabel])
 }
+
+// Validate given hostIP IP belongs to the current host
+// adapted from github.com/kubernetes/kubernetes/pkg/kubelet/kubelet_node_status.go
+func validateHostIP(hostIP net.IP) error {
+	if hostIP.To4() == nil {
+		return fmt.Errorf("hostIP must be a valid IP address")
+	}
+	if hostIP.IsLoopback() {
```

Comment:
Not? I’m wondering why 127.0.0.1 should not work...

Reply:
We want to choose an IP with as close a heuristic as the kubelet as possible. the kubelet filters loopback, so I did. I can't really think of a reason it wouldn't work though, but I think I'd like to keep it there.